## Population genetics of *Porites cf. lobata* across the Palauan archipelago 
This repo contains the data and scripts associated with Rivera et al. 2022 (in revison), Palau’s warmest reefs harbor a thermally tolerant coral lineage that thrives across different habitats; a paper on thermal tolerance and population genetics of *Porites cf. lobata* corals across the Palauan archipelago. 

A preprint of this manuscript is available [here](https://www.researchsquare.com/article/rs-1190526/v1). Please note, that in the pdf version of the preprint, figures are not properly rendered for some reason. The figures are fine in the html version linked above. Also note, that the manuscript has undergone some substantial revisions since the preprint, though the main conclusions of the manuscript remain unchanged. 

### Repository Organization 
The repository is separated into a population genetic folder, which contains SNPs, microsatellite, and growth data; and a 
temperature folder, which contains temperature data and scripts. 

Within both folders there is an .RData file that can be loaded directly to avoid some of the initial data processing and begin 
with data analyses. 

All the input data that would be needed to run these scripts is included in the data folder. Intermediate files that can be 
generated by running the scripts with the included input data are not included (though most of these are in the .RData file for convenience). Please note that the vcf files were uploaded using Git Large File Storage (LFS), if downloading the repository you will need to install git LFS prior to downloading in order to actually retrive the files. See more information about Git LFS [here](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-git-large-file-storage). 

Final figures that are in the manuscript and supplementary data are included in the respective figures and tables folders for easy reference (most of these are under the Coral_Data folder).

### Coral colony collection 
Between 2011-18, we collected tissue from 543 *Porites cf. lobata* colonies using a hammer and chisel while on SCUBA (Table S2). Tissue was preserved in RNAlater™ (Invitrogen, Waltham, MA), incubated overnight at 4°C, and frozen at -20°C (N=329), or frozen directly at -80°C (N=20), or preserved in 95% ethanol and frozen at -20°C (N=194) until DNA extraction. Colonies were sampled haphazardly within each site, across 13 sites (Fig. 2), based on morphological characteristics of Porites lobata detailed in Veron (2000).  

The site “Ngermid” has been referred to as “Nikko Bay” in previous publications. We use “Ngermid” here as that is the name preferred by Palauan natives. The “Outer Taoch” site contained samples from three outer reef locations: Airai (GPS coordinates: 7.33210, 134.56020, N= 5), Rael Dil (7.24990, 134.45073, N= 3), and a fringing reef (7.27193, 134.38115, N=30) immediately outside of Taoch Bay. The coordinates for this last site were used for mapping because most of the samples in this group are from this location. Due to the presence of various lineages within our dataset, population genetic metrics (e.g., F<sub>ST</sub>) were not calculated by collection site, so this choice should have no bearing on results.  

![Map of sampling sites](https://github.com/hrivera28/Palau_porites/blob/main/Coral_data/Popgen_R/figures/fig2/Palau_Sites_All_data_donuts_inferred_linages.png)

### DNA extraction, microsatellite, and SNP genotyping
A thawed ~1 mm2 piece of coral was homogenized into a fine powder using a new standard safety razor blade sterilized with ethyl alcohol and flamed. The homogenate was processed using the Qiagen® DNeasy Blood and Tissue DNA extraction kit according to the manufacturer instructions, with a modified Proteinase K incubation of at least 24 hours. Negative controls (N=5) without any coral tissue added were included every 70 samples and subjected to all the same downstream processing and analyses. 

We amplified 14 microsatellite markers with fluorescently labeled primers developed for *P. cf. lobata* by Baums et al. (2015). PCR settings were: (1) initial denaturation at 94°C for 5 min, (2) 35 cycles of 94°C for 20 seconds, annealing at 52, 54, or 56°C (plex-dependent) for 20 seconds, 72°C for 30 seconds, and (3) final extension for 30 minutes at 72°C. The Pennsylvania State University Nucleic Acid Facility measured fragments on an ABI 3730 (GeneScan) with a LIZ-500 internal size standard.

A subset of samples with sufficient quality extracted DNA were then processed for RAD sequencing. Genomic DNA concentrations were standardized using a Qubit™ 2.0 fluorometer (Invitrogen) to 20 ng/ul. A total of 50 ul per sample was sent to Floragenex (Portland, Oregon) for single enzyme RAD library preparation with PstI enzyme digestion. Each sample was identified by a unique 10 nucleotide barcode. Samples (N=185) were sequenced as 100 base pair single end reads across 6 lanes of an Illumina Hiseq 4000™ using v4 chemistry at the University of Oregon Genomics Core facility. 

### RAD-seq processing and SNP-calling (script: Coral_data/snp_filtering.sh)
Raw reads were processed using the ‘process_radtags’ module of Stacks v.1.4665, allowing for up to three mismatches in the sample barcode (this was the maximum number of mismatches at which the barcodes remained unique). Reads with low quality scores (PHRED<10) across a sliding window of 15% of the read length were discarded. We retained 78% of the original reads. Average sequencing depth was 8.5 million reads per sample. Three samples replicated within the plate showed 1.5-2-fold variability in sequencing depth. One sample which had an unusually high number of reads (>35 million) was discarded. 

For read mapping and SNP calling, we used the dDocent v.2.3.7 pipeline66 with a Porites lutea or P. lutea symbiont draft genome obtained from the REFUGE 2020 database (http://refuge2020.reefgenomics.org/) as reference (for coral and symbionts SNPs, respectively). dDocent clustered reads based on >95% similarity using CD-HIT v.4.6.867, mapped reads to the reference using the MEM algorithm of BWA v.0.7.1768 with a match score of 1, mismatch score of 3, and gap-open penalty of 4, and called SNPs using FreeBayes v.1.1.069 with default values (E=3, m=PHRED 10, q=PHRED10, -V, and using the sampling sites as the populations designations). The resulting ‘TotalRawSNPs.vcf’ file was filtered using vcftools v.0.1.1570 and vcffilter (https://github.com/jameshicks/vcffilter) following the suggestions in the dDocent manual, with a final thinning (-thin option in vcftools to keep only SNPs more than 150 bp apart, e.g. only one SNP per rad tag) to obtain a final set of 12,761 bi-allelic SNPs in 146 retained samples for the coral host and 218 bi-allelic SNPs in 137 individuals for the symbiont. These final vcf files are avaialble in Coral_data/Popgen_R/input_files. 







